### PR TITLE
Fixed Auth, Route Guards, and Server Connection

### PR DESCRIPTION
**Pull request by djpeach**

### CODE REVIEW

File: ionic-client/src/App.js [hunk 1]
```diff
@@ -35,24 +35,12 @@ const App = () => (
...
-          <Route exact path={routes.lists.detail} render={props => {
-            return !!firebase.auth().currentUser ? <List {...props} /> : <Login/>;
-          }}/>
-          <Route exact path={routes.suggestions.create} render={props => {
-            return !!firebase.auth().currentUser ? <NewSuggestion {...props} /> : <Login/>;
-          }}/>
-          <Route exact path={routes.more} render={props => {
-            return !!firebase.auth().currentUser ? <More {...props} /> : <Login/>;
-          }}/>
+        <Route exact path={routes.auth.login} component={Login}/>
```
Comment (trentspi) [inline ionic-client/src/App.js]: Add a tab here (spacing)

File: ionic-client/src/components/common/AuthedComponent.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import firebase from 'firebase';
+import { Redirect } from 'react-router';
+import routes from '../../conf/routes';
+
+function authedComponent(WrappedComponent) {
+  return class extends React.Component {
+    render() {
+      // Wraps the input component in a container, without mutating it. Good!
+      return !!firebase.auth().currentUser ? (<WrappedComponent {...this.props}/>) : (<Redirect to={routes.auth.login} />);
```
Comment (trentspi) [inline ionic-client/src/components/common/AuthedComponent.js]: Nice!

File: ionic-client/src/index.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ import {InMemoryCache} from 'apollo-cache-inmemory';
 import keys from './conf/secret-keys'
 import App from './App';
 
-const httpLink = new HttpLink({uri: 'http://localhost:9000/graphql'});
+const httpLink = new HttpLink({uri: 'http://bucketlist.us-east-2.elasticbeanstalk.com/graphql'});
```
Comment (trentspi) [inline ionic-client/src/index.js]: Nice!

File: ionic-client/src/App.js (hunk 1 above)
Comment (djpeach) [reply ionic-client/src/App.js]: damn forgot to run prettier

**Pull request merged**